src/core_atmosphere/physics/physics_wrf/module_cu_ntiedtke.F

```diff
@@ -353,42 +356,61 @@ subroutine cu_ntiedtke(                                    &
    do j=jts,jte
 
 ! --------------- compute zi and zl -----------------------------------------
+!$acc parallel vector_length(128)
+!$acc loop gang vector
       do i=its,ite
+        cu_act_flag(i,j)=.true.
         zi(i,kts)=0.0
       enddo
+!$acc end parallel
 !
       do k=kts,kte
+!$acc parallel vector_length(128)
+!$acc loop gang vector
         do i=its,ite
           zi(i,k+1)=zi(i,k)+dz8w(i,k,j)
         enddo
+!$acc end parallel
       enddo
 !
+!$acc parallel vector_length(128)
+!$acc loop gang vector collapse(2)
       do k=kts,kte
         do i=its,ite
           zl(i,k)=0.5*(zi(i,k)+zi(i,k+1))
         enddo
       enddo
+!$acc end parallel
 
 ! --------------- end compute zi and zl -------------------------------------
+!$acc parallel vector_length(128)
+!$acc loop gang vector
       do i=its,ite
         slimsk(i)=int(abs(xland(i,j)-2.))
       enddo
+!$acc end parallel
 
+!$acc parallel vector_length(128)
       do i=its,ite
          dx2d(i) = dx(i,j)
       enddo
+!$acc end parallel
 
+!$acc parallel vector_length(128)
+!$acc loop gang vector collapse(2)
       do k=kts,kte
-        kp=k+1
+!!!     kp=k+1
         do i=its,ite
-          dot(i,k)=-0.5*g*rho3d(i,k,j)*(w(i,k,j)+w(i,kp,j))
+          dot(i,k)=-0.5*g*rho3d(i,k,j)*(w(i,k,j)+w(i,k+1,j))
         enddo
       enddo
 
-      pp = 0
+!!!   pp = 0
+!$acc loop gang vector collapse(2) private(zz)
       do k=kts,kte
-        zz = kte-pp
+!!!     zz = kte-pp
         do i=its,ite
+          zz = kte+1-k
```

Comment:
This looks incorrect unless kts is 1 to begin with.
I think this ought to be
zz=kte-(k-kts)